cobbler/actions/buildiso/__init__.py

```diff
@@ -231,7 +326,74 @@ def calculate_grub_name(self, distro) -> str:
             f' Found: "{str(matches.values())}"'
         )
 
-    def __prepare_buildisodir(self, buildisodir: str = "") -> str:
+    def _write_isolinux_cfg(
+        self, cfg_parts: List[str], output_dir: pathlib.Path
+    ) -> None:
+        """Write isolinux.cfg.
+
+        :param cfg_parts: List of str that is written to the config, joined by newlines.
+        :param output_dir: pathlib.Path that the isolinux.cfg file is written into.
+        """
+        output_file = output_dir / "isolinux.cfg"
+        self.logger.info("Writing %s", output_file)
+        with open(output_file, "w") as f:
+            f.write("\n".join(cfg_parts))
+
+    def _write_grub_cfg(self, cfg_parts: List[str], output_dir: pathlib.Path) -> None:
+        """Write grub.cfg.
+
+        :param cfg_parts: List of str that is written to the config, joined by newlines.
+        :param output_dir: pathlib.Path that the grub.cfg file is written into.
+        """
+        output_file = output_dir / "grub.cfg"
+        self.logger.info("Writing %s", output_file)
+        with open(output_file, "w") as f:
+            f.write("\n".join(cfg_parts))
+
+    def _create_esp_image_file(self, tmpdir: str) -> str:
+        esp = pathlib.Path(tmpdir) / "efi"
+        mkfs_cmd = ["mkfs.fat", "-C", str(esp), "3528"]
+        rc = utils.subprocess_call(mkfs_cmd, shell=False)
+        if rc != 0:
+            self.logger.error("Could not create ESP image file")
+            raise Exception  # TODO: use proper exception
+        return str(esp)
+
+    @contextlib.contextmanager
+    def _mount_esp(self, esp_path: str):
```

Comment:
Mounting is something we can't do in rootless containers. Instead of creating and mounting the fat image, I want to try a different approach based on https://unix.stackexchange.com/questions/629492/create-and-populate-fat32-filesystem-without-mounting-it

Requiring privileges just to create an UEFI-bootable ISO should be our last resort.

Reply:
Let's do it later. It's not super hard but requires changing a few things. We need to first create the directory structure, then copy it into the ESP with `mcopy`, then remove the directory structure in the tempdir. The tempdir can also conflict with `EFI/BOOT` in `buildisodir`, unless we want to use the same contents for both (inside the ESP and outside).

Anyway, this is something we need to improve, but it does not have to be today.